server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -1313,6 +1314,31 @@ def percentage(
     ) -> StepContext:
         raise NotImplementedError(f"[{self.DIALECT}] percentage is not implemented")
 
+    def rank(
+        self: Self,
+        *,
+        builder: "QueryBuilder",
+        prev_step_table: str,
+        columns: list[str],
+        step: "RankStep",
+    ) -> StepContext:
+        col_field: Field = Table(prev_step_table)[step.value_col]
+        new_col_name = step.new_column_name or f"{step.value_col}_rank"
```

Comment:
```suggestion
        new_col_name = step.new_column_name or f"{step.value_col}_RANK"
```
not a big deal, just to match if possible others ? : https://github.com/ToucanToco/weaverbird/blob/master/server/src/weaverbird/backends/pandas_executor/steps/rank.py#L13 for pandas and https://github.com/ToucanToco/weaverbird/blob/master/server/src/weaverbird/backends/mongo_translator/steps/rank.py#L68 on mongo

Reply:
This would lead to inconsistent behaviour between different backends: Redshift converts everything to lowercase https://github.com/ToucanToco/weaverbird/actions/runs/5658104353/job/15328658382

Reply:
But if we decide that consistency between backends is important I can investigate on why redshift behaves differently :slightly_smiling_face:

Reply:
Nope, IMO, i think we can keep it like that then.
it's a restriction from the engine, let's not put too much effort on a small gain like this, WDYT ?

Reply:
Agreed :smile: